### DISCUSSION TITLE
Changing scenario immediately in one table.

### DISCUSSION THREAD

**Discussion by n-h-tuan** [CityScope/CS_cityscopeJS#139] [Q&A]
Hi @RELNO, when working with CityScope_JS, I got an idea from my team that they want to change multiple scenarios (which are set up in the source code) but not want to get the delay between sending the data to CityIO and get it back via websocket. In other word, they want to change the scenario, including the pie chart, also the brix data in local, not through the server. So, can we do anything similarly like that? Thank you @RELNO 😀
![question1](https://user-images.githubusercontent.com/35844864/120140130-11400c80-c204-11eb-8821-1c32c0141eed.gif)

**Comment by RELNO** [comment]
Hi @nguyenledly thanks for the question. 
To answer, we should first define a few concepts of CityScope, and then explain what is possible. 

1. CityScope is based on **modules** that compute different urban analytics based on changes to a shared protocol (`GEOGRID`)
2. When a module is done computing, its output is sent to cityIO server so that both the frontend as well as other modules would be able to use this data for visualization or further computation (for example, a noise module might impact another mobility module)
3. CityIO is thus central for this communication. I believe you referred to it as websokcets, but it's in fact a REST api. 
4. 'Local' can mean a few different things: running a copy of cityIO on premise / running all module computation on the same machine / running the whole stack from the browser. Can you clarify to which you refer?  @yasushisakai  can share insights about running an instance of cityIO locally, though I'm not sure if the benefits worth the complexity. 
5.  As we've discussed in our recent call, the usage of Geojsons as the only input ('scenarios') is not aligned with how CityScope is built. If the intention is indeed to only interact with a handful of predefined Geojsons, simple`POST` requests to `https://cityio.media.mit.edu/api/table/***YOUR TABLE***/geojson`could do the trick (such as https://cityio.media.mit.edu/api/table/test/geojson)
6. However, (5) will only allow you to upload and display the geojsons on the frontend. None of our current modules is designed to work with such input data, but instead are build around `GEOGRID` and `GEOGRIDDATA`. 

Hope this helps,
Ariel
